addons/mrp/report/mrp_report_bom_structure.py

```diff
@@ -20,15 +20,15 @@ def _get_report_values(self, docids, data=None):
                 if data and data.get('childs'):
                     doc = self._get_pdf_line(bom_id, product_id=product_variant_id, qty=float(data.get('quantity')), child_bom_ids=json.loads(data.get('childs')))
                 else:
-                    doc = self._get_pdf_line(bom_id, product_id=product_variant_id, unfolded=True)
+                    doc = self._get_pdf_line(bom_id, product_id=product_variant_id, qty=float(data.get('quantity')), unfolded=True)
                 doc['report_type'] = 'pdf'
                 doc['report_structure'] = data and data.get('report_type') or 'all'
                 docs.append(doc)
             if not candidates:
                 if data and data.get('childs'):
                     doc = self._get_pdf_line(bom_id, qty=float(data.get('quantity')), child_bom_ids=json.loads(data.get('childs')))
                 else:
-                    doc = self._get_pdf_line(bom_id, unfolded=True)
+                    doc = self._get_pdf_line(bom_id,  qty=float(data.get('quantity')), unfolded=True)
```

Comment:
Could you use data.get('quantity', 0.0) instead.

Reply:
Changes done